### PR TITLE
feat: add support for numbers in multicall

### LINKED ISSUE #2245: Add support for numbers in multicall
### Description

It seems counterintuitive not to allow raw integers inputs in the mutlicall, right now it's not supported and only string inputs are accepted

**Example:**

```
[[call]]
call_type = "deploy"
class_hash = "0x076e94149fc55e7ad9c5fe3b9af570970ae2cf51205f8452f39753e9497fe849"
inputs = []
id = "map_contract"
unique = false

[[call]]
call_type = "invoke"
contract_address = "map_contract"
function = "put"
inputs = ["0x123", "234"]
```

### Desired Behavior

We can allow the `inputs` to be array of numbers:

```
...
[[call]]
call_type = "invoke"
contract_address = "map_contract"
function = "put"
inputs = [123, 0x234]
```

Also `contract_address` and `class_hash` can be numbers,
Example:

```
[[call]]
call_type = "deploy"
class_hash = 0x076e94149fc55e7ad9c5fe3b9af570970ae2cf51205f8452f39753e9497fe849"
inputs = []
id = "map_contract"
unique = false
```

### Context
- [Path to the file that needs altering](https://github.com/foundry-rs/starknet-foundry/blob/bc608465bc72b1c84648bbae6558182deccd9e35/crates/sncast/src/starknet_commands/multicall/run.rs#L4)
- [Multicall docs](https://foundry-rs.github.io/starknet-foundry/starknet/multicall.html)
- [Numbers in TOML format](https://toml.io/en/)

### PR DESCRIPTION
**Pull request by kfastov**
Closes #2245

## Introduced changes

- Added new enum type `Input` with variants `String` and `Number(i64)` to support both string and numeric inputs in TOML files
- Updated `DeployCall` and `InvokeCall` structs to use `Vec<Input>` instead of `Vec<String>` for their `inputs` field
- Modified `parse_inputs` function to handle both string and numeric input variants
- Added `#[serde(untagged)]` attribute to enable seamless deserialization of both input types from TOML

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

## Other notes
- Numbers larger than 2^64 are not supported by TOML format: https://toml.io/en/v1.0.0#integer
- However, numbers outside i64 range cannot be supported currently due to an [issue](https://github.com/toml-rs/toml/issues/705) in the toml parser

### CODE REVIEW

**Comment by Arcticae** [comment]
@kfastov can you please also test the case where the number exceeds the limit (so we make sure it fails with a reasonable message)?

**Comment by Arcticae** [comment]
@kfastov bump

**Comment by Arcticae** [comment]
@kfastov are you still working on this? Please let me know so i can reassign if you're not :)

**Comment by kfastov** [comment]
Hi @Arcticae 
Sorry for delayed reply, I had a multi-day transfer with no internet, I am ready to finish this PR now, and I'll do it today or tomorrow :)

**Comment by Arcticae** [comment]
Thx for the update  👍

**Review by Arcticae — APPROVED** [review]
cc @cptartur can you assign someone to take a second look?

File: docs/src/starknet/multicall.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ After running `sncast multicall run --path file.toml --fee-token strk`, a declar
 > The example above demonstrates the use of the `id` property in a deploy call, which is then referenced as the `contract address` in an invoke call.
 Additionally, the `id` can be referenced in the inputs of deploy and invoke calls 🔥
 
+> 💡 **Info**
+> Inputs can be either strings (like "0x123") or numbers (like 234).
+
+> 📝 **Note**
+> For numbers larger than 2^63 - 1 (that can't fit into i64), use string format (e.g., "9223372036854775808") due to TOML parser limitations.
```
Comment (franciszekjob) [inline docs/src/starknet/multicall.md:47]: nit:
```suggestion
> For numbers larger than 2^63 - 1 (that can't fit into `i64`), use string format (e.g., `"9223372036854775808"`) due to TOML parser limitations.
```

File: docs/src/starknet/multicall.md [hunk 2]
```diff
@@ -40,6 +40,12 @@ After running `sncast multicall run --path file.toml --fee-token strk`, a declar
 > The example above demonstrates the use of the `id` property in a deploy call, which is then referenced as the `contract address` in an invoke call.
 Additionally, the `id` can be referenced in the inputs of deploy and invoke calls 🔥
 
+> 💡 **Info**
+> Inputs can be either strings (like "0x123") or numbers (like 234).
```
Comment (franciszekjob) [inline docs/src/starknet/multicall.md:44]: nit
```suggestion
> Inputs can be either strings (like `"0x123"`) or numbers (like `234`).
```

**Review by franciszekjob — APPROVED** [review]
Left only two nits, looking good 👍

**Comment by kfastov** [comment]
@franciszekjob Addressed the nits, please check

**Review by franciszekjob — APPROVED** [review]
LGTM

**Comment by Arcticae** [comment]
@kfastov thx for your contribution!

**Pull request merged**